AmplifyTests/CategoryTests/Auth/AuthCategoryConfigurationTests.swift

```diff
@@ -0,0 +1,367 @@
+    /// - When:
+    ///    - I coose one plugin and call one of the Auth API
+    /// - Then:
+    ///    - API should complete without error for one plugin
+    ///
+    func testCanUseSpecifiedPlugin() throws {
+        let plugin1 = MockAuthCategoryPlugin()
+        let methodShouldNotBeInvokedOnDefaultPlugin =
+            expectation(description: "test method should not be invoked on default plugin")
+        methodShouldNotBeInvokedOnDefaultPlugin.isInverted = true
+        plugin1.listeners.append { message in
+            if message == "changePassword" {
+                methodShouldNotBeInvokedOnDefaultPlugin.fulfill()
+            }
+        }
+        try Amplify.add(plugin: plugin1)
+
+        let plugin2 = MockSecondAuthCategoryPlugin()
+        let methodShouldBeInvokedOnSecondPlugin =
+            expectation(description: "test method should be invoked on second plugin")
+        plugin2.listeners.append { message in
+            if message == "changePassword" {
+                methodShouldBeInvokedOnSecondPlugin.fulfill()
+            }
+        }
+        try Amplify.add(plugin: plugin2)
+
+        let config = AuthCategoryConfiguration(
+            plugins: [
+                "MockAuthCategoryPlugin": true,
+                "MockSecondAuthCategoryPlugin": true
+            ]
+        )
+
+        let amplifyConfig = AmplifyConfiguration(auth: config)
+
+        try Amplify.configure(amplifyConfig)
+        _ = try Amplify.Auth.getPlugin(for: "MockSecondAuthCategoryPlugin")
+            .changePassword(currentPassword: "current",
+                            newPassword: "new",
+                            options: nil,
+                            listener: nil)
+        waitForExpectations(timeout: 1.0)
+    }
+
+    /// Test if we get error when trying default plugin when multiple plugin added.
+    ///
+    /// - Given: Amplify configured with multiple auth plugin
+    /// - When:
+    ///    - I try to invoke an API with default plugin
+    /// - Then:
+    ///    - Should throw an exception
+    ///
+    func testPreconditionFailureInvokingWithMultiplePlugins() throws {
+        let plugin1 = MockAuthCategoryPlugin()
+        try Amplify.add(plugin: plugin1)
+
+        let plugin2 = MockSecondAuthCategoryPlugin()
+        try Amplify.add(plugin: plugin2)
+
+        let config = AuthCategoryConfiguration(
+            plugins: [
+                "MockAuthCategoryPlugin": true,
+                "MockSecondAuthCategoryPlugin": true
+            ]
+        )
+
+        let amplifyConfig = AmplifyConfiguration(auth: config)
+
+        try Amplify.configure(amplifyConfig)
+
+        let exception: BadInstructionException? = catchBadInstruction {
+            _ = Amplify.Auth.changePassword(currentPassword: "current",
+                                            newPassword: "new",
+                                            listener: nil)
+        }
+        XCTAssertNotNil(exception)
```

Comment:
is there something about the exception other than nil, that we can check to make sure we are throwing a specific exception?

Reply:
We are casting the error to 'BadInstructionException'. So that way we are trying to assert that it is of that type by checking its not nil.